scilpy/tracking/tracker.py

```diff
@@ -0,0 +1,173 @@
+# -*- coding: utf-8 -*-
+import numpy as np
+
+from scilpy.tracking.tools import sample_distribution
+from scilpy.tracking.utils import TrackingDirection
+
+
+class abstractPropagator(object):
+
+    def __init__(self, tracker, step_size):
+        self.step_size = step_size
+        self.tracker = tracker
+
+    def propagate(self, pos, v_in):
+        pass
+
+    def getValidDirection(self, pos, v_in):
+        is_valid_direction = True
+        v_out = self.tracker.get_direction(pos, v_in)
+        if v_out is None:
+            is_valid_direction = False
+            v_out = v_in
+
+        return is_valid_direction, v_out
+
+
+class rk1Propagator(abstractPropagator):
+
+    """
```

Comment:
Could you add the same brief description for rk2 and rk4

Reply:
Done in 703949e